src/main/java/Task.java

```diff
@@ -0,0 +1,24 @@
+import java.util.*;
+
+public class Task {
+    public static final String descriptionPrefix = "description: ";
+    private String description;
+    private boolean important;
+    List<String> pastDescription = new ArrayList<>(); // a list of past descriptions
+
+    public Task(String d) {
+      this.description = d;
+      if (!d.isEmpty())
+          this.important = true;
+    }
+
+    public String getAsXML() { return "<task>"+description+"</task>"; }
+
+    /**
+     * Print the description as a string.
+     */
+    public void printingDescription(){ System.out.println(this); }
```

Comment:
Usage of the **this** keyword only when a field is shadowed by a method or constructor parameter. Here the **this** a little ambiguous.